### PR TITLE
Implemented Fraction

### PR DESCRIPTION
**Pull request by parth121101**
Fixes: #3400

### CODE REVIEW

File: src/lfortran/semantics/ast_common_visitor.h [hunk 1]
```diff
@@ -748,6 +748,7 @@ class CommonVisitor : public AST::BaseVisitor<Derived> {
         {"ishftc", {IntrinsicSignature({"i", "shift"}, 2, 2)}},
         {"ichar", {IntrinsicSignature({"C", "kind"}, 1, 2)}},
         {"char", {IntrinsicSignature({"I", "kind"}, 1, 2)}},
+        {"fraction", {IntrinsicSignature({"X"}, 1, 1)}},
```
Comment (HarshitaKalani) [inline src/lfortran/semantics/ast_common_visitor.h:751]: I think we need not register this here if the number of arguments is 1.

File: src/libasr/intrinsic_func_registry_util_gen.py [hunk 2]
```diff
@@ -382,7 +382,13 @@
         {
             "args": [("real",)],
             "return": "int32",
-        },
+        }
```
Comment (HarshitaKalani) [inline src/libasr/intrinsic_func_registry_util_gen.py:385]:
```suggestion
        },
```
We usually put a comma here.

File: src/libasr/intrinsic_func_registry_util_gen.py [hunk 3]
```diff
@@ -382,7 +382,13 @@
...
+    ],    
+    "Fraction": [
+        {
+            "args": [("real",)],
+            "ret_type_arg_idx": 0
+        }
```
Comment (HarshitaKalani) [inline src/libasr/intrinsic_func_registry_util_gen.py:391]:
```suggestion
        },
```
Similarly here

**Review by certik — APPROVED** [review]

**Comment by certik** [comment]
Conflicts must be resolved.

File: integration_tests/intrinsics_173.f90 [hunk 4]
```diff
@@ -0,0 +1,34 @@
+program intrinsics_171
+    real :: x
+    real :: y
+    real(8) :: z
+    real(8) :: w
```
Comment (certik) [inline integration_tests/intrinsics_173.f90:5]: Use `dp` not 8.

File: integration_tests/intrinsics_173.f90 [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    if (fraction(x) - 0.57004 > 1e-5) error stop
+
+    print *, fraction(y)
+    if (fraction(y) - 0.50000 > 1e-5) error stop
+
+    print *, fraction(z)
+    if (fraction(z) - (-0.63749998807907104_8)  > 1e-8) error stop
+
+    print *, fraction(w)
+    if (fraction(w) - (0.0000000000000000_8) > 1e-5) error stop
```
Comment (certik) [inline integration_tests/intrinsics_173.f90:21]: Use dp.

**Comment by certik** [comment]
Otherwise good.

**Comment by certik** [comment]
Please submit a new PR with the `dp` change.

**Comment by parth121101** [comment]
> Please submit a new PR with the `dp` change.

Sure @certik

**Pull request merged**